pallets/connectors/src/lib.rs

```diff
@@ -0,0 +1,507 @@
+			// 		domain: address.domain,
+			// 	}
+			// 	.into_account_truncating(),
+			// 	amount,
+			// 	false,
+			// )?;
+
+			Self::do_send_message(
+				who,
+				Message::Transfer {
+					pool_id,
+					tranche_id,
+					amount,
+					domain: address.clone().domain,
+					destination: address.clone().address,
+				},
+				address.domain,
+			)?;
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub(crate) fn now() -> Moment {
+			T::Time::now().as_secs()
+		}
+
+		/// Send the `message` to the given domain.
+		pub fn do_send_message(
+			fee_payer: T::AccountId,
+			message: MessageOf<T>,
+			domain: Domain,
+		) -> DispatchResult {
+			let Router::Xcm(xcm_domain) =
+				<DomainRouter<T>>::get(domain.clone()).ok_or(Error::<T>::MissingRouter)?;
+
+			let contract_call = Self::encoded_contract_call(message.encode());
+			let ethereum_xcm_call =
+				Self::encoded_ethereum_xcm_call(xcm_domain.clone(), contract_call);
+
+			pallet_xcm_transactor::Pallet::<T>::transact_through_sovereign(
+				T::Origin::root(),
+				// The destination to which the message should be sent
+				Box::new(xcm_domain.location),
+				fee_payer,
+				// The currency in which we want to pay fees
+				CurrencyPayment {
+					currency: Currency::AsCurrencyId(xcm_domain.fee_currency),
+					fee_amount: None,
+				},
+				// The call to be executed in the destination chain
+				ethereum_xcm_call,
+				OriginKind::SovereignAccount,
+				TransactWeights {
+					// Specify a conservative max weight
+					transact_required_weight_at_most: 8_000_000_000,
+					overall_weight: None,
+				},
+			)?;
+
+			Self::deposit_event(Event::MessageSent { message, domain });
+
+			Ok(())
+		}
+
+		/// Build the encoded `ethereum_xcm::transact(eth_tx)` call that should
+		/// request to execute `evm_call`.
+		///
+		/// * `xcm_domain` - All the necessary info regarding the xcm-based domain
+		/// where this `ethereum_xcm` call is to be executed
+		/// * `evm_call` - The encoded EVM call calling ConnectorsXcmRouter::handle(msg)
+		pub fn encoded_ethereum_xcm_call(
+			xcm_domain: XcmDomain<CurrencyIdOf<T>>,
+			evm_call: Vec<u8>,
+		) -> Vec<u8> {
+			let mut encoded: Vec<u8> = Vec::new();
+
+			encoded.append(&mut xcm_domain.ethereum_xcm_transact_call_index.clone());
+			encoded.append(
+				&mut xcm_primitives::EthereumXcmTransaction::V1(
+					xcm_primitives::EthereumXcmTransactionV1 {
+						gas_limit: U256::from(80_000),
+						fee_payment: xcm_primitives::EthereumXcmFee::Auto,
+						action: pallet_ethereum::TransactionAction::Call(
+							xcm_domain.contract_address,
+						),
+						value: U256::zero(),
+						input: evm_call,
+						access_list: None,
+					},
+				)
+				.encode(),
+			);
+
+			encoded
+		}
+
+		/// Return the encoded contract call, i.e, ConnectorsXcmRouter::handle(encoded_msg).
+		pub fn encoded_contract_call(encoded_msg: Vec<u8>) -> Bytes {
+			let contract = xcm_router_contract();
+			let encoded_contract_call = contract
+				.function("handle")
```

Comment:
I guess `handle` should always be the same, but wouldn't hurt to make it a constant?

Reply:
yes good idea

Reply:
Done